src/autosuggest/autosuggest-input.tsx

```diff
@@ -0,0 +1,265 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { Ref, useRef, useState, useImperativeHandle, useEffect } from 'react';
+
+import Dropdown from '../internal/components/dropdown';
+
+import { FormFieldValidationControlProps, useFormFieldContext } from '../internal/context/form-field-context';
+import { BaseComponentProps, getBaseProps } from '../internal/base-component';
+import { useUniqueId } from '../internal/hooks/use-unique-id';
+import {
+  BaseKeyDetail,
+  fireCancelableEvent,
+  fireNonCancelableEvent,
+  getBlurEventRelatedTarget,
+  NonCancelableEventHandler,
+} from '../internal/events';
+import InternalInput from '../input/internal';
+import { BaseChangeDetail, BaseInputProps, InputKeyEvents, InputProps } from '../input/interfaces';
+import { getFocusables } from '../internal/components/focus-lock/utils';
+import { ExpandToViewport } from '../internal/components/dropdown/interfaces';
+import { InternalBaseComponentProps } from '../internal/hooks/use-base-component';
+import { KeyCode } from '../internal/keycode';
+import styles from './styles.css.js';
+
+export interface AutosuggestInputProps
+  extends BaseComponentProps,
+    BaseInputProps,
+    InputKeyEvents,
+    FormFieldValidationControlProps,
+    ExpandToViewport,
+    InternalBaseComponentProps {
+  ariaControls?: string;
+  ariaActivedescendant?: string;
+  dropdownExpanded?: boolean;
+  dropdownContent?: React.ReactNode;
+  dropdownFooter?: React.ReactNode;
+  dropdownWidth?: number;
+  onCloseDropdown?: NonCancelableEventHandler<null>;
+  onDelayedInput?: NonCancelableEventHandler<BaseChangeDetail>;
+  onPressArrowDown?: () => void;
+  onPressArrowUp?: () => void;
+  onPressEnter?: () => boolean;
+}
+
+export interface AutosuggestInputRef extends InputProps.Ref {
+  focusNoOpen(): void;
```

Comment:
Does not have to be in scope of this PR, but we might expose this feature to public API, because it is useful for our [validation pattern](https://cloudscape.design/patterns/general/validation/). When autosuggest dropdown opens on focus, it covers error text below the field.

If this happens, we probably want to make it an argument instead of a separate function:

```js
autosuggestRef.current.focus(); // opens dropdown
autosuggestRef.current.focus({ preventDropdown: true }); // dropdown stays closed
```

Reply:
I like the idea and can do it as a follow-up and with having an api-review doc.